src/lib/apis/convection.ts

```diff
@@ -2,14 +2,20 @@ import urljoin from "url-join"
 import { assign } from "lodash"
 import fetch from "./fetch"
 import config from "config"
+import { APIOptions } from "lib/loaders/api"
 
-const { CONVECTION_API_BASE } = config
+const { CONVECTION_API_BASE, CONVECTION_TOKEN } = config
 
-export default (path, accessToken, fetchOptions = {}) => {
+export default (path, accessToken, fetchOptions: APIOptions = {}) => {
   const headers = { Accept: "application/json" }
-  if (accessToken) assign(headers, { Authorization: `Bearer ${accessToken}` })
+  // appToken goes in the headers!!!
+  const { appToken, ...optionsForFetch } = fetchOptions
```

Comment:
i _think_ you don't need to handle/worry about `appToken` here.

`accessToken` -> this is the `X-Access-Token` header of the request hitting MP. If present, should be propagated and take precedence.
`CONVECTION_TOKEN` -> long-lived app token signed by Convection, set on MP's ENV. When `accessToken` isn't present, use this one.

I think that should cover it.

`appToken` -> this is the `X-Xapp-Token` header of the request hitting MP. In some cases we want to propagate this around, but for the use case of a logged-out user, hitting Metaphysics -> Convection, you _want_ to use `CONVECTION_TOKEN`. Otherwise, this might be Force's (or some other client app) xapp token.

Reply:
Thank youuu!